### PR TITLE
Deploy Obscuro Gateway

### PR DESCRIPTION
**Pull request by zkokelj**
### Why this change is needed

We would like to have a Github action that will deploy Obscuro Gateway on Azure

### What changes were made as part of this PR

- Dockerfile
- `container_build.sh` and `container_run.sh`
- new Github action in: `.github/workflows/manual-deploy-obscuro-gateway.yml`

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/obscuronet/obscuro-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [ ] PR checks reviewed and performed

### CODE REVIEW

**Review by otherview — APPROVED** [review]
Looks good, would request to making those changes before merging 👍

File: .github/workflows/manual-deploy-obscuro-gateway.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Deploys Obscuro Gateway on Azure for Testnet
+# Builds the Faucet network image, pushes the image to dockerhub and starts the Obscuro Gateway on Azure
+
+name: '[M] Deploy Obscuro Gateway Testnet'
+run-name: '[M] Deploy Obscuro Gateway Testnet ( ${{ github.event.inputs.testnet_type }} )'
+on:
+  workflow_dispatch:
+# todo: Add options to add configuration options
```
[the remark this answers is not in the document]
Comment (zkokelj) [reply .github/workflows/manual-deploy-obscuro-gateway.yml:8]: I was thinking about adding an option to deploy for mainnet etc, but we would probably like to create a separate github action for that. (Will delete the comment)

File: .github/workflows/manual-deploy-obscuro-gateway.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: 'Set up Docker'
+        uses: docker/setup-buildx-action@v1
+
+      - name: 'Set Obscuro gateway build tag'
+        run: |
+          echo "OBSCURO_GATEWAY_BUILD_TAG=testnetobscuronet.azurecr.io/obscuronet/obscuro_gateway_testnet:latest" >> $GITHUB_ENV
```
[the remark this answers is not in the document]
Comment (zkokelj) [reply .github/workflows/manual-deploy-obscuro-gateway.yml:21]: Added an if statement + inputs under workflow dispatch to enable us to choose if we are deploying to testnet or dev-testnet

File: tools/walletextension/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,11 @@
+FROM golang:1.18-alpine
+
+RUN apk add build-base
+
+RUN mkdir /obscuro-gateway
+WORKDIR /obscuro-gateway
+COPY . .
```
[the remark this answers is not in the document]
Comment (zkokelj) [reply tools/walletextension/Dockerfile:7]: todo added

File: tools/walletextension/container_build.sh [hunk 4]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (zkokelj) [reply tools/walletextension/container_build.sh:1]: removed

File: tools/walletextension/container_run.sh [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+# Stop and remove any running container, and then star
+echo "Force stopping any existing container ... "
+docker rm -f  obscuro_gateway_testnet 2>/dev/null
+
+echo "Starting Obscuro Gateway..."
+docker run -p 3000:"${port}" --name=obscuro_gateway_testnet \
+    --detach \
+    --entrypoint ./tools/walletextension/main/obscuro-gateway \
+     "obscuronet/obscuro_gateway_testnet:latest" \
```
[the remark this answers is not in the document]
Comment (zkokelj) [reply tools/walletextension/container_run.sh:47]: Fixed. Is it ok to use `obscuronet/obscuro_gateway_testnet:latest`?
Or shoud I add `azurecr.io` as in faucet?

File: .github/workflows/manual-deploy-obscuro-gateway.yml [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+      - name: 'Set Obscuro Gateway build tag for dev-testnet'
+        if: ${{ github.event.inputs.testnet_type == 'dev-testnet' }}
+        run: |
+          echo "OBSCURO_GATEWAY_BUILD_TAG=testnetobscuronet.azurecr.io/obscuronet/dev_obscuro_gateway_testnet:latest" >> $GITHUB_ENV
+
+      - name: 'Login to Azure docker registry'
+        uses: azure/docker-login@v1
+        with:
+          login-server: testnetobscuronet.azurecr.io
+          username: ${{ secrets.REGISTRY_USERNAME }}
```
Comment (otherview) [inline .github/workflows/manual-deploy-obscuro-gateway.yml:41]: This was changed due to output secret clashes.

```suggestion
          username: testnetobscuronet
```

**Review by otherview — APPROVED** [review]

**Pull request merged**